### PR TITLE
Fix some Any subclassing in tqdm

### PR DESCRIPTION
**Pull request by Avasam**
Fixes ~~4~~ 2 Any subclassing issues by replicating the types used in external modules.
~~The `rich` one isn't complete as it was starting to get quite complex (I don't wanna re-implement the entire library's types either!)~~

Ref: #9491

### CODE REVIEW

File: stubs/tqdm/tqdm/keras.pyi [hunk 1]
```diff
@@ -1,10 +1,31 @@
 from _typeshed import Incomplete
-from typing import Any
-from typing_extensions import TypeAlias
 
 __all__ = ["TqdmCallback"]
 
-_Callback: TypeAlias = Any  # Actually tensorflow.keras.callbacks.Callback
+# keras.callbacks.Callback
```
Comment (AlexWaygood) [inline stubs/tqdm/tqdm/keras.pyi:5]:
```suggestion
# tensorflow.python.keras.callbacks.Callback
```

**Review by AlexWaygood — COMMENTED** [review]
The `keras` and `dask` changes look great. Mind splitting the `rich` changes off into a separate PR? They're a little more complex and my brain's going a little fuzzy thinking about nominal vs structural subtyping 😅

File: stubs/tqdm/tqdm/keras.pyi (hunk 1 above)
Comment (Avasam) [reply stubs/tqdm/tqdm/keras.pyi:5]: Weirdly enough, it's actually using keras, and if that fails, it fallsback to tensorflow... which uses keras (probably does some extra in that case idk)

Comment (AlexWaygood) [reply stubs/tqdm/tqdm/keras.pyi:5]: Oh good point, ignore me!

**Comment by Avasam** [comment]
> The `keras` and `dask` changes look great. Mind splitting the `rich` changes off into a separate PR? They're a little more complex and my brain's going a little fuzzy thinking about nominal vs structural subtyping 😅

If it feels too complex for our needs, I can always make more things `Incomplete`. We mostly need the structure, detailed types are bonuses.

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Comment by AlexWaygood** [comment]
> > The `keras` and `dask` changes look great. Mind splitting the `rich` changes off into a separate PR? They're a little more complex and my brain's going a little fuzzy thinking about nominal vs structural subtyping 😅
> 
> If it feels too complex for our needs, I can always make more things `Incomplete`. We mostly need the structure just to fix the Any subclassing, detailed types are bonuses.

Maybe we could start off with something simpler for now? We can always refine it in the future :)

**Pull request merged**